components/eventForm/SubmitButton.tsx

```diff
@@ -0,0 +1,77 @@
+import { FontScaleFactors } from "../../lib/FontScale"
+import React from "react"
+import { Alert, StyleSheet, Text, TouchableOpacity, View } from "react-native"
+import { useEventFormContext } from "./EventForm"
+import { eventEditInputFromFormValues } from "./EventFormValues"
+
+/**
+ * Props from `EventFormSubmitButton`.
+ */
+export type EventFormSubmitButtonProps = {
+  label: string
+}
+
+/**
+ * Submit button for `EventForm`.
+ */
+export const EventFormSubmitButton = ({
+  label
+}: EventFormSubmitButtonProps) => {
+  const submitButtonTapped = useSubmit()
+  const color = useEventFormContext().watch("color")
+  const canSubmit = !!submitButtonTapped
+  return (
+    <View style={{ opacity: canSubmit ? 1 : 0.5 }}>
+      <TouchableOpacity
+        onPress={() => submitButtonTapped?.()}
+        disabled={!canSubmit}
+        style={{
+          ...styles.container,
+          backgroundColor: color,
+          shadowColor: color
+        }}
+      >
+        <Text
+          maxFontSizeMultiplier={FontScaleFactors.xxxLarge}
+          disabled={!canSubmit}
+          style={styles.button}
+        >
+          {label}
+        </Text>
+      </TouchableOpacity>
+    </View>
+  )
+}
+
+const useSubmit = () => {
+  const { formState, submit, watch, hasEdited } = useEventFormContext()
+  const updateInput = eventEditInputFromFormValues(watch())
+  const canSubmit = !!updateInput && !formState.isSubmitting && hasEdited
+  if (!canSubmit) return undefined
+  return async () => {
+    try {
+      await submit(updateInput)
+    } catch {
+      // TODO: - Should we just forward the actual error message from the thrown error here?
```

Comment:
Good question, coordinate with Nico and Tristan on the kinds of errors you want to receive. Although if we support multiple languages in the future I don't think we could easily translate those errors from the backend.

Reply:
We wouldn't just use the error from the backend directly, rather we would translate an error code/identifier into an error message on the frontend which would be in the proper language